Code-Chef/MEDEL.cpp

```diff
@@ -0,0 +1,44 @@
+#include <iostream>
+using namespace std;
+int del(int a[],int x,int n){
+	int i;
+	if(x<=n-2){
+		for(i=x;i<n;i++)
+			a[i]=a[i+1];
+	}
+}
+int main() {
+	int t;
+	cin>>t;
+	while(--t>=0){
+	    int n,i,t;
+	    cin>>n;
+	    int a[n];
+	    t=n;
+	    for(i=0;i<n;i++)cin>>a[i];
+	    while(t>=3){
+	    	if(a[0]>=a[1] && a[0]>=a[2]){
+	    		if(a[1]>=a[2])
+	    			del(a,1,t);
+	    		else
+	    			del(a,2,t);
+			}
+			else if(a[1]>=a[2]){
+				if(a[0]>=a[2])
+					del(a,0,t);
+				else
+					del(a,2,t);
+			}
+			else{
+				if(a[1]>=a[0])
+					del(a,1,t);
+				else
+					del(a,0,t);
+			}
+			t--;
+		}
+	    for(i=0;i<2;i++)cout<<a[i]<<" ";
+		cout<<endl;
+	}
+	return 0;
+}
```

Comment:
I suggest you to update the name of the variables so that anyone can understand how your code is solving the problem or if that is too much to ask, then add an editorial for your solution, so that anybody can understand the code when they refer your code and editorial back and forth.

Reply:
@invictus1599 In the problem mendel, you can solve the problem simply by finding the largest and the smallest element in the array, because when you play around with some test-cases you will see that the after removing the middle element from the 3-element sub-array which we are considering every-time, we will be left with only 2 elements after the traversing the whole array.  
And these 2 elements which we are left with are the smallest and the largest element in the array.  
Only the think you need to keep the order in which element occurs in the array i.e.  
```C
if(max_val_index > min_val_index) {
    printf("%"PRIu8" %"PRIu8"\n", min_val, max_val);
} else {
    printf("%"PRIu8" %"PRIu8"\n", max_val,min_val);
}
```
Refer to the solution: https://github.com/strikersps/Competitive-Programming/tree/master/Code-Chef/Medel